script/lib/s3put.js

```diff
@@ -0,0 +1,38 @@
+/* eslint-disable camelcase */
+const AWS = require('aws-sdk');
+const fs = require('fs');
+const path = require('path');
+AWS.config.update({ region: 'us-west-2' });
+const s3 = new AWS.S3({ apiVersion: '2006-03-01' });
+
+const args = require('minimist')(process.argv.slice(2));
+
+let { bucket, prefix = '/', key_prefix = '', grant, _: files } = args;
+if (prefix && !prefix.endsWith(path.sep)) prefix = path.resolve(prefix) + path.sep;
+
+function filenameToKey (file) {
+  file = path.resolve(file);
+  if (file.startsWith(prefix)) file = file.substr(prefix.length);
+  return key_prefix + file.replace(path.sep, '/');
+}
+
+let anErrorOccurred = false;
+function next (done) {
+  const file = files.shift();
+  if (!file) return done();
+  s3.upload({
```

Comment:
Where does s3 get authenticated?

Reply:
It automatically pulls the environment variables: https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/EnvironmentCredentials.html